### PR TITLE
chore: convert codebase to ESM

### PR DESCRIPTION
**Pull request by digitalsadhu**
BREAKING CHANGE: JavaScript API is now ESM only and no longer supports CJS

### CODE REVIEW

**Review by leftieFriele — APPROVED** [review]
Looks good to me, ESM 🔥  one thing I was wondering about, should we update all the outdated dependencies as well while doing this breaking change?

File: commands/publish.js [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-const fetch = require('node-fetch');
-const ora = require('ora');
-const chalk = require('chalk');
-const {
-    helpers: { configStore },
-} = require('@eik/common');
-const PublishPackage = require('../classes/publish/package/index');
-const {
+import { join } from 'path';
+import fetch from 'node-fetch';
```
Comment (wkillerud) [inline commands/publish.js:2]: We could use this opportunity to drop `node-fetch` as a dependency and use the built-in. Would mean a mininum of Node 18, but that's fine IMO.

File: test/fixtures/client-with-bare-imports.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
-
-'use strict';
-
+/* eslint-disable no-undef */
+// @ts-nocheck
 import scrollIntoView from 'scroll-into-view-if-needed';
 
 const main = () => {
-    // eslint-disable-next-line no-undef
-    const node = document.getElementById('hero');
```
Comment (wkillerud) [inline test/fixtures/client-with-bare-imports.js:9]: Not that this is the part under test, but I think this missing line is what `eslint` is complaining about 😅

File: utils/hash/compare.js [hunk 3]
```diff
@@ -10,4 +8,4 @@
  *
  * @example hash.compare('a1b22c23d24e25f4g33a123b23c34', 'a1b22c23d24e25f4g33a123b23c34');
  */
-module.exports = (hash1, hash2) => hash1 === hash2;
+export default (hash1, hash2) => hash1 === hash2;
```
Comment (wkillerud) [inline utils/hash/compare.js:11]: A bit of an aside, but does this need to be its own function? 😅

File: classes/index.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
 'use strict';
```
Comment (wkillerud) [inline classes/index.js:1]: Nit: there's a few places where `use strict` can be removed

**Comment by wkillerud** [comment]
> should we update all the outdated dependencies as well while doing this breaking change?

+1, would be nice to do a sweep of the open PRs and bump deps that would be a breaking change.

File: utils/hash/compare.js (hunk 3 above)
Comment (digitalsadhu) [reply utils/hash/compare.js:11]: Nah but I'm not touching the structure in this PR. Thats a black hole to be avoided when ESMing 😆

**Comment by digitalsadhu** [comment]
Might be worth either merging this into a "next" and then updating dept, then merging into main or doing dep updates as part of this PR to avoid multiple breaking changes

File: commands/publish.js (hunk 1 above)
Comment (digitalsadhu) [reply commands/publish.js]: Removed in latest commit

File: classes/index.js (hunk 4 above)
Comment (digitalsadhu) [reply classes/index.js]: Done!

**Review by wkillerud — APPROVED** [review]

**Comment by digitalsadhu** [comment]
This PR is now against next and I'll merge it into that and continue work updating dependencies from there

**Pull request merged**